src/config.rs

```diff
@@ -1,63 +1,69 @@
-                        theme
-                    }
-                }
-            };
-
-            let task_store = if !data_path.exists() {
-                TaskStore::default()
-            } else {
-                let data_contents = fs::read_to_string(&data_path);
-                match data_contents {
-                    Ok(file) => serde_json::from_str::<TaskStore>(&file)?,
-                    Err(_) => {
-                        let tasks: Vec<Task> = vec![];
-                        fs::write(&data_path, serde_json::to_string(&tasks)?)?;
-                        TaskStore::default()
-                    }
+use std::{error::Error, fs};
+
+use crate::{app::TaskStore, theme::Theme};
+
+const DIR: &str = "dotodo";
+
+const CONFIG_FILE: &str = "config.yml";
+const DATA_FILE: &str = "data.json";
+
+pub fn get_data() -> (Theme, TaskStore) {
+    let config_local_dir = dirs::config_local_dir();
+    let data_local_dir = dirs::data_local_dir();
+
+    let mut theme = Theme::default();
+
+    if let Some(dir) = config_local_dir {
+        let config_path = dir.join(DIR).join(CONFIG_FILE);
+
+        if config_path.exists() {
+            if let Ok(config) = fs::read_to_string(&config_path) {
+                if let Ok(theme_config) = serde_yaml::from_str::<Theme>(&config) {
+                    theme = theme_config;
                 }
-            };
-            Ok((theme, task_store))
+            }
         }
-        None => {
-            println!("Not found");
-            Ok((Theme::default(), TaskStore::default()))
+    }
+
+    let mut task_store = TaskStore::default();
+
+    if let Some(dir) = data_local_dir {
+        let data_path = dir.join(DIR).join(DATA_FILE);
+
+        if data_path.exists() {
+            if let Ok(data) = fs::read_to_string(&data_path) {
+                if let Ok(task_store_data) = serde_json::from_str::<TaskStore>(&data) {
+                    task_store = task_store_data;
+                }
+            }
         }
     }
+
+    (theme, task_store)
 }
 
 // FIX: proper error handling, pring data out if cannout save.
 pub fn save_data(theme: &Theme, task_store: &TaskStore) -> Result<(), Box<dyn Error>> {
-    let dotodo_path = dirs::home_dir().unwrap().join(".config/dotodo/");
-
-    fs::create_dir_all(dotodo_path)?;
-    fs::write(
-        dirs::home_dir().unwrap().join(".config/dotodo/data.json"),
-        serde_json::to_string(task_store)?,
-    )?;
-    fs::write(
-        dirs::home_dir().unwrap().join(".config/dotodo/config.yml"),
-        serde_yaml::to_string(theme)?,
-    )?;
+    let config_local_dir = dirs::config_local_dir();
+    let data_local_dir = dirs::data_local_dir();
+
+    if let Some(dir) = config_local_dir {
+        let config_dir = dir.join(DIR);
+
+        // NOTE: It's technically possible for OS-specific utils that this fn calls
+        // to fail if the path already exists.
+        fs::create_dir_all(config_dir.clone())?;
```

Comment:
It would be nice for this to have an error message if it fails if possible.

Reply:
I think we should add some kind of an error pop-up. For now, maybe just put some `eprintln!()`'s?
Would be nice to also save some log (e.g. `log.txt`) with error messages (or also actions, not sure). For now, one could just redirect the stderr to a file.

Reply:
Yea, that should be fine for now.